pallets/identity/src/lib.rs

```diff
@@ -125,6 +125,43 @@ decl_storage! {
         /// All authorizations that an identity/key has given. (Authorizer, auth_id -> authorized)
         pub AuthorizationsGiven: double_map hasher(blake2_256) Signatory, blake2_256(u64) => Signatory;
     }
+    add_extra_genesis {
+        config(identities): Vec<(T::AccountId, IdentityId, IdentityId, u64)>;
+        build(|config: &GenesisConfig<T>| {
+            for &(ref master_account_id, did_issuer, did, ref expiry) in &config.identities {
+                // Direct storage change for registering the DID and providing the claim
+                let master_key = AccountKey::try_from(master_account_id.encode()).unwrap();
+                assert!(!<DidRecords>::exists(did), "Identity already exist");
+                <MultiPurposeNonce>::mutate(|n| *n += 1_u64);
+                <Module<T>>::link_key_to_did(&master_key, SignatoryType::External, did);
+                let record = DidRecord {
+                    master_key,
+                    ..Default::default()
+                };
+                <DidRecords>::insert(&did, record);
+
+                // Add the claim data for the CustomerDueDiligence type claim
+                use wasm_timer::SystemTime;
+
+                let now = SystemTime::now();
```

Comment:
I am not sure if this will work either. Can you please share a link to the code where it is being used in the timestamp pallet?

Reply:
https://github.com/paritytech/substrate/blob/a201f34963dbd6fb7a266a5d6f0b5660ca0db34b/primitives/timestamp/src/lib.rs#L96

Reply:
I don't think that code is meant to be used for consensus. Seems to me that It just returns the current time on the local system and it will be different on different systems.

Reply:
I'll run a few tests on my local to confirm/dismiss my suspicion. Please don't merge it till then.

Reply:
Yes, that code doesn't mean for consensus as it is submitted as inherent in every block. That's how only the block timestamp gets updated within the substrate blockchain. Anyway please check for double sure. Other than that I am not sure for any easy way to get the deterministic time.

Reply:
Yeah, this is working as I expected. i.e. This is returning the current system time. That means it is generating different values every time you purge the chain. I don't think we want that. That will cause sync issues between nodes as they will have different states. The genesis block should be fixed.

Reply:
@satyamakgec You are not fetching the time from the variable set in the inherent though. You are fetching it directly from the system. That can cause different nodes to produce different results and that can cause a consensus failure. Either please fetch the time from the inherent or set a fixed time.